tests/inspection/test_lazy_wheel.py

```diff
@@ -0,0 +1,324 @@
+from __future__ import annotations
+
+import re
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+from typing import Protocol
+from typing import Tuple
+from urllib.parse import urlparse
+
+import httpretty
+import pytest
+import requests
+
+from requests import codes
+
+from poetry.inspection.lazy_wheel import HTTPRangeRequestUnsupported
+from poetry.inspection.lazy_wheel import InvalidWheel
+from poetry.inspection.lazy_wheel import metadata_from_wheel_url
+
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from httpretty.core import HTTPrettyRequest
+
+    from tests.types import FixtureDirGetter
+
+    HttPrettyResponse = Tuple[int, Dict[str, Any], bytes]  # status code, headers, body
+    HttPrettyRequestCallback = Callable[
+        [HTTPrettyRequest, str, Dict[str, Any]], HttPrettyResponse
+    ]
+
+    class RequestCallbackFactory(Protocol):
+        def __call__(
+            self,
+            *,
+            accept_ranges: str | None = "bytes",
+            negative_offset_error: tuple[int, bytes] | None = None,
+        ) -> HttPrettyRequestCallback: ...
+
+
+NEGATIVE_OFFSET_AS_POSITIVE = -1
+
+
+def build_head_response(
+    accept_ranges: str | None, content_length: int, response_headers: dict[str, Any]
+) -> HttPrettyResponse:
+    response_headers["Content-Length"] = content_length
+    if accept_ranges:
+        response_headers["Accept-Ranges"] = accept_ranges
+    return 200, response_headers, b""
+
+
+def build_partial_response(
+    rng: str,
+    wheel_bytes: bytes,
+    response_headers: dict[str, Any],
+    *,
+    negative_offset_as_positive: bool = False,
+) -> HttPrettyResponse:
+    status_code = 206
+    response_headers["Accept-Ranges"] = "bytes"
+    total_length = len(wheel_bytes)
+    if rng.startswith("-"):
+        # negative offset
+        offset = int(rng)
+        if negative_offset_as_positive:
+            # some servers interpret a negative offset like "-10" as "0-10"
+            start = 0
+            end = min(-offset, total_length - 1)
+            body = wheel_bytes[start : end + 1]
+        else:
+            start = total_length + offset
+            if start < 0:
+                # wheel is smaller than initial chunk size
+                return 200, response_headers, wheel_bytes
+            end = total_length - 1
+            body = wheel_bytes[offset:]
+    else:
+        # range with start and end
+        rng_parts = rng.split("-")
+        start = int(rng_parts[0])
+        end = int(rng_parts[1])
```

Comment:
```suggestion
        start, end = map(int, rng_parts)
```
Or if `rng_parts` isn't guaranteed to be of length 2:
```suggestion
        start, end, *_ = map(int, rng_parts)
```